scripts/quests/otherAreas/An_Affable_Adamantking.lua

```diff
@@ -0,0 +1,321 @@
+            {
+                [61] = function(player, csid, option, npc)
+                    npcUtil.giveKeyItem(player, xi.ki.ORCISH_SEEKER_BATS)
+                    quest:setVar(player, 'Prog', 4)
+                end,
+
+                [62] = function(player, csid, option, npc)
+                    if
+                        npcUtil.popFromQM(player, npc, {
+                            qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET,
+                            qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET + 1,
+                            qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET + 2,
+                            qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET + 3,
+                            qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET + 4,
+                            qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET + 5,
+                            qulunDomeID.mob.AFFABLE_ADAMANTKING_OFFSET + 6,
+                        }, { claim = true, hide = 0 })
+                    then
+                        GetNPCByID(qulunDomeID.npc.BEASTMENS_BANNER):setLocalVar('engaged', 1)
+                    end
+                end,
+
+                [63] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 6)
+                end,
+
+                [64] = function(player, csid, option, npc)
+                    if player:getFreeSlotsCount() > 0 then
+                        player:confirmTrade()
+                        npcUtil.giveItem(player, xi.item.DAVHUS_BARBUT)
+                        player:setCharVar('affableZone', 1)
+                        player:setVar('BstHeadGearQuest_Conquest', NextConquestTally())
+                        quest:complete(player) -- quest is marked complete, additional reward upon zoning w/ CS
+                    else
+                        player:messageSpecial(qulunDomeID.text.ITEM_CANNOT_BE_OBTAINED, xi.item.DAVHUS_BARBUT)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_COMPLETED
+        end,
+
+        [xi.zone.QULUN_DOME] =
+        {
+            onZoneIn =
+            {
+                function(player, prevZone)
+                    local pos = player:getPos()
+                    if
+                        pos.x >= 0 and
+                        pos.y >= 20 and
+                        pos.z >= 55 and
+                        pos.x <= 6 and
+                        pos.y <= 25 and
+                        pos.z <= 65 and
+                        player:getCharVar('affableZone') == 1 and
+                        player:getEquipID(xi.slot.HEAD) == xi.item.DAVHUS_BARBUT
+                    then
+                        return 65
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [65] = function(player, csid, option, npc)
+                    if player:getFreeSlotsCount() > 0 then
```

Comment:
This can be simplified, as I believe giveItem will also send the appropriate message if the player cannot receive the item.  if npcUtil.giveItem(player..) then set charvar to 0

Reply:
Refactored to use giveitem🤘